core/server/worker/src/main/java/alluxio/worker/block/BlockStoreMeta.java

```diff
@@ -12,195 +12,91 @@
 package alluxio.worker.block;
 
 import alluxio.collections.Pair;
-import alluxio.worker.block.meta.StorageDir;
-import alluxio.worker.block.meta.StorageTier;
 
-import com.google.common.base.Preconditions;
+import net.jcip.annotations.ThreadSafe;
 
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.LinkedList;
 import java.util.List;
 import java.util.Map;
 
-import javax.annotation.concurrent.ThreadSafe;
-
 /**
- * This class holds the metadata information of a block store.
- * <p>
- * TODO(bin): Use proto buf to represent this information.
+ * Interface for the block store meta in Alluxio.
  */
-@ThreadSafe
-public final class BlockStoreMeta {
-  // TODO(bin): The following two fields don't need to be computed on the creation of each
-  // {@link BlockStoreMeta} instance.
-
-  /** Mapping from storage tier alias to capacity bytes. */
-  private final Map<String, Long> mCapacityBytesOnTiers = new HashMap<>();
-
-  /** Mapping from storage tier alias to used bytes. */
-  private final Map<String, Long> mUsedBytesOnTiers = new HashMap<>();
-
-  /** Mapping from storage tier alias to capacity bytes. */
-  private final Map<String, List<Long>> mBlockIdsOnTiers;
-
-  /** Mapping from storage dir tier and path to total capacity. */
-  private final Map<Pair<String, String>, Long> mCapacityBytesOnDirs = new HashMap<>();
-
-  /** Mapping from storage dir tier and path to used bytes. */
-  private final Map<Pair<String, String>, Long> mUsedBytesOnDirs = new HashMap<>();
+public interface BlockStoreMeta {
 
   /**
-   * Factory method to return a BlockStoreMeta instance without blockIds.
-   *
-   * @param manager the BlockMetadataManager
-   * @return BlockStoreMeta instance
+   * Factory for {@link BlockStoreMeta}.
    */
-  public static BlockStoreMeta getBlockStoreMeta(BlockMetadataManager manager) {
-    return new BlockStoreMeta(manager, false);
-  }
+  @ThreadSafe
+  class Factory{
+
+    private Factory() {}
+
+    /**
+     * Factory method to return a BlockStoreMeta instance without blockIds.
+     *
+     * @param manager the BlockMetadataManager
+     * @return BlockStoreMeta instance
+     */
+    public static BlockStoreMeta getBlockStoreMeta(BlockMetadataManager manager) {
```

Comment:
NIT:  I would rename this to `create`, as it is in `BlockStoreMeta.Factory` and it is obvious that what it creates is an instance of `BlockStoreMeta`